src/components/Waveform/WaveformTypes.ts

```diff
@@ -10,6 +10,7 @@ interface BaseWaveform {
   containerStyle?: StyleProp<ViewStyle>;
   waveColor?: string;
   mode: StaticOrLive;
+  onWaveformLoadStateChange?: (state: boolean) => void;
```

Comment:
I think this will only be needed for static waveforms, not base waveforms, as this method should only be available for static. Live waveforms do not need to indicate whether the extraction process is running or done, as we do not do that for live ones.

Reply:
It works for both static as well as live waveforms

Reply:
Done, Please Check